ambient-poly.py

```diff
@@ -36,49 +35,66 @@ def longPoll(self):
         States that data is updated every 5 or 30 minutes (so which is it?)
         """
         LOGGER.info('Connecting to Ambient Weather server')
-        LOGGER.info(self.api_key)
-        LOGGER.info(self.mac_address)
-
-        http = urllib3.PoolManager()
-        #c = httplib.HTTPSConnection("api.ambientweather.net")
+        # #LOGGER.info(self.api_key)
+        # #LOGGER.info(self.mac_address)
 
         # TODO: Make the path as part of the config procesing so that
         # we only do it once.
         # how to limit this to one entry?
-        path_str = 'http://api.ambientweather.net/'
-        path_str += 'v1/devices/' + self.mac_address + '?apiKey='
-        path_str += self.api_key 
-
-        path_str += '&applicationKey=5644719c27144e3a9b0341238344c7a4bf11a7c5023f4b4cbc1538b834b80037'
-        path_str += '&limit=1'
-        LOGGER.info(path_str)
-
-        c = http.request('GET', path_str)
-        #c.request("GET", path_str)
-
-        #response = c.getresponse()
-        #response = c.data
-        #print response.status, response.reason
-        print (c.status, c.reason)
-        #data = response.read()
-        #c.close()
-        awdata = json.loads(c.data.decode('utf-8'))
-
-        # deserialize data into an object?
-        LOGGER.info(awdata[0])
+        # TODO: May want to make the app key user provided?
+        awAPI = 'https://api.ambientweather.net/'
+        awAPI += 'v1/devices/' + self.mac_address + '?apiKey=' + self.api_key
+        awAPI += '&applicationKey=5644719c27144e3a9b0341238344c7a4bf11a7c5023f4b4cbc1538b834b80037'
+        awAPI += '&limit=1'
+        #LOGGER.info(awAPI)
+
+        r = requests.get(awAPI)
+        #LOGGER.info(r.json())
+        awdata = r.json()
         d = awdata[0]
 
-        # TODO: calculate additional data values
-        # pressure trend
-        # heat index
-        # windchill
-        # rain rate
-
+        '''
```

Comment:
I like the idea of adding the additional sensors, but I'm not sure about adding the nodes in the polling function. I do like the idea of it dynamically checking but I'm thinking it might be better to refactor the existing code first and separate the query from the set driver code.  Then the query function could be called during discover at startup and add the nodes there based on what data is present in the data.  That would also allow the other existing nodes to be created based on the data present as well.

Reply:
They are dynamic in two ways.
1. They are added automatically to AW when the sensor is powered on.  There is no add/remove process.  Generally speaking people add a sensor and never remove it unless the batteries die and they never replace them.
2. Because they are RF I've seen mine appear/disappear due to different conditions.  Technically the sensor still exists therefore the node should still exist but AW is not reporting any data for it at that point in time.  So this one is where leaving the node makes sense as the data will reappear or should anyways.

As for a status value.  The only "status" is whether there is data available in AW or not.  There's no other method/capability to poll the sensor itself or even to get battery level from these.  They are either on or off and either reporting or not.

I'm all up for making things more configurable.

Reply:
Ok, so maybe they shouldn't be considered that dynamic. Issue #2 seems more like a HW bug or something related to local conditions and not really intended behavior for the sensors.   

Too bad they don't have battery status, that would be handy :)  But my thought was that if data was present when polled, mark the sensor status as "on-line" and if the data wasn't present mark it as "off-line".   That way it would be possible to detect stale sensor data.   But if this isn't really expected behavior for the sensor then I'm not sure I'd bother adding it.

But a lot of this comes down to how you are planning or expecting this data to be used in the ISY. I don't have a Ambient stations, but I would expect the sensor data to be used to trigger actions like if sensor02.tempf > 100 turn on fan.   Thus, if you can't expect to get current/valid data for sensor02, it's not going to be too useful.

The discover function can be run at any time, by the user.  The main controller node has a command to kick off this process.   It may make the most sense to have the user re-run discover (or restart the node server) when new sensor devices are  installed.   Running a query from discover may mean we have to delay the poll based query, but that should be pretty easy to do, just set a flag and skip the next poll query if it is set.  I guess we may have to also wait to do the discovery query.  I'm kind of thinking about this as I type.  So maybe the best solution would be to have discover set a flag and in LongPoll check the flag. If set pass the queried data to a create_nodes() function, then continue with the set drivers. 
```Python
def discover:
     run_discovery = True

def LongPoll:
    awdata = query()
    if run_discovery:
      create_nodes(data)
      run_discovery = False

    foreach node:
       setDriver(awdata[])
````
I'd probably break this down into a commit that refactors the existing node creation into a separate function and move it to happen on the Poll. So basically like the pseudo code above but without adding the new sensor nodes.  Then a second commit to add in the new sensor nodes. Does this make sense or am I missing something?